src/gt4py/cartesian/gtc/daceir.py

```diff
@@ -252,9 +253,9 @@ def intersection(cls, axis, first, second):
         second_start = second.start if second.start is not None else first.start
         second_end = second.end if second.end is not None else first_end.end
 
-        assert (first_start <= second_end and first_end >= second_start) or (
-            second_start <= first_end and second_end >= first_start
-        )
+        if (first_start <= second_end and first_end >= second_start) or (
+            second_start <= first_end and second_end >= first_start) :
+            warnings.warn(f"extension is NOT monotonically increasing!")
```

Comment:
Please give context. The assert makes sure that the intersection is not empty. I agree that it could make sense to represent an empty DomainInterval, but then I would not give a warning.

Reply:
I don't mind whether we remove this warning message or not, I give it as a warning is to use it as a reminder for the users or app developers the stencil is void, this is needed for pyFV3 corner update stencils as it is void stencils when there is no corner info to update.  sorry for delayed response.

Reply:
@havogt yeah, I remove the warning message, just release the assert statement and allow the void stencil, please review it again

Reply:
I still need more context to review this. Can you add a testcase that exercises this new code path? Alternatively, if @FlorianDeconinck wants to take the responsibility and just approve, it's ok for me.

Reply:
yeah, will add a testcase